samples/bluetooth/hci_uart/src/main.c

```diff
@@ -154,12 +142,18 @@ static void bt_uart_isr(struct device *unused)
 				continue;
 			}
 
-			switch (type) {
-			case H4_CMD:
-				buf = h4_cmd_recv(&remaining);
+			buf = bt_buf_get_tx(BT_BUF_ACL_OUT, K_NO_WAIT, &type,
+					    sizeof(type));
+			if (!buf) {
+				return;
+			}
+
+			switch (bt_buf_get_type(buf)) {
```

Comment:
This looks broken to me.

Firstly, it looks super odd to pass explicitly `BT_BUF_ACL_OUT` as the first parameter to `bt_buf_get_tx()`. when you don't know what type the packet actually is. Secondly, that function uses this first parameter to call `bt_buf_set_type()`, so you'll never get anything back in the switch statement except `ACL_OUT`.

Reply:
Actually, `bt_buf_get_tx()` calls `bt_buf_set_type()` twice, the second time overwriting the first, which confused me. Can you update the code so that it only sets the type once?

Reply:
And add a new `BT_BUF_AUTO` enum for this purpose. Whenever HCI raw H4 isn't enabled passing `BT_BUF_AUTO` should return error, or perhaps better, assert.

Reply:
Will update the code with BT_BUF_AUTO, could it perhaps be called BT_BUF_H4 which is onlt defined when BT_HCI_RAW_H4 is selected?

Reply:
Done.